packages/tools/examples/segmentBidirectionalTool/findLargestBidirectional.ts

```diff
@@ -0,0 +1,218 @@
+import { cache } from '@cornerstonejs/core';
+import { adaptersRT } from '@cornerstonejs/adapters';
+import * as cornerstoneTools from '@cornerstonejs/tools';
+
+import vtkImageMarchingSquares from '@kitware/vtk.js/Filters/General/ImageMarchingSquares';
+import vtkDataArray from '@kitware/vtk.js/Common/Core/DataArray';
+import vtkImageData from '@kitware/vtk.js/Common/DataModel/ImageData';
+import { vec3 } from 'gl-matrix';
+
+const vtkUtils = {
+  vtkImageMarchingSquares,
+  vtkDataArray,
+  vtkImageData,
+};
+
+const EPSILON = 1e-2;
+
+export default function contourAndFindLargestBidirectional(segmentations) {
+  const { generateContourSetsFromLabelmap } = adaptersRT.Cornerstone3D.RTSS;
+
+  console.time('contour');
+  const contours = generateContourSetsFromLabelmap({
+    segmentations,
+    cornerstoneCache: cache,
+    cornerstoneToolsEnums: cornerstoneTools.Enums,
+    vtkUtils,
+  });
+  console.timeEnd('contour');
+
+  if (!contours?.length || !contours[0].sliceContours.length) {
+    return;
+  }
+
+  const { representationData, segments = [0, 1] } = segmentations;
+  const { Labelmap } = cornerstoneTools.Enums.SegmentationRepresentations;
+  const { volumeId: segVolumeId } = representationData[Labelmap];
+
+  const segmentIndex = segments.findIndex((it) => !!it);
+  if (segmentIndex === -1) {
+    return;
+  }
+  segments[segmentIndex].segmentIndex = segmentIndex;
+  return findLargestBidirectional(
+    contours[0],
+    segVolumeId,
+    segments[segmentIndex]
+  );
+}
+
+function createIsInSegment(segVolumeId: string, segment: number) {
+  // Get segmentation volume
+  const vol = cache.getVolume(segVolumeId);
+  if (!vol) {
+    console.warn(`No volume found for ${segVolumeId}`);
+    return;
+  }
+
+  const segData = vol.imageData.getPointData().getScalars().getData();
+  const width = vol.dimensions[0];
+  const pixelsPerSlice = width * vol.dimensions[1];
+
+  return (pointI, pointJ) => {
+    const point = vec3.add(vec3.create(), pointI, pointJ).map((it) => it / 2);
+    const ijk = vol.imageData.worldToIndex(point as vec3).map(Math.round);
+    const [i, j, k] = ijk;
+    const index = i + j * width + k * pixelsPerSlice;
+    const value = segData[index];
+    return value === segment;
+  };
+}
+
+function findLargestBidirectional(contours, segVolumeId, segment) {
+  const { sliceContours } = contours;
+  const { segmentIndex } = segment;
+  let maxBidirectional;
+  const isInSegment = createIsInSegment(segVolumeId, segmentIndex);
+  for (const sliceContour of sliceContours) {
+    const startTimeMs = Date.now();
+    const bidirectional = createBidirectionalForSlice(
+      sliceContour,
+      maxBidirectional,
+      isInSegment
+    );
+    const durationMs = Date.now() - startTimeMs;
+    if (!bidirectional) {
+      if (durationMs > 250) {
+        console.log('Skipped slice bidirectional took', durationMs);
+      }
+      continue;
```

Comment:
so this is like if it takes more than 250ms we skip it? seems odd

Reply:
No, if it takes mroe than 250 ms, we log a message indicating it is slow.  I could adjust the others to be like that, even with warnings.